### ISSUE TITLE
Test Leader Elector Suite

### ISSUE THREAD

**Issue by jackstine** [solo-io/gloo#7988]
The [logs are here](https://storage.googleapis.com/solo-public-build-logs/logs.html?buildid=ec62ecc0-dd64-48ad-a414-b7db82310136).

```bash
Step #1 - "run-tests": Step #7 - "run-tests": [38;5;9m• [FAILED] [0.016 seconds][0m
Step #1 - "run-tests": Step #7 - "run-tests": [0mLeader Startup Action [38;5;9m[1m[It] StartupAction executed once, if elected[0m
Step #1 - "run-tests": Step #7 - "run-tests": [38;5;243m/workspace/gloo/pkg/bootstrap/leaderelector/leader_test.go:59[0m
Step #1 - "run-tests": Step #7 - "run-tests": 
Step #1 - "run-tests": Step #7 - "run-tests":   [38;5;9m[FAILED] Failed after 0.014s.
Step #1 - "run-tests": Step #7 - "run-tests":   Unexpected error:
Step #1 - "run-tests": Step #7 - "run-tests":       <*internal.asyncPolledActualError | 0xc000353c10>: {
Step #1 - "run-tests": Step #7 - "run-tests":           message: "The function passed to Consistently failed at /workspace/gloo/pkg/bootstrap/leaderelector/leader_test.go:69 with:\nExpected\n    <uint64>: 2\nto equal\n    <uint64>: 1",
Step #1 - "run-tests": Step #7 - "run-tests":       }
Step #1 - "run-tests": Step #7 - "run-tests":       The function passed to Consistently failed at /workspace/gloo/pkg/bootstrap/leaderelector/leader_test.go:69 with:
Step #1 - "run-tests": Step #7 - "run-tests":       Expected
Step #1 - "run-tests": Step #7 - "run-tests":           <uint64>: 2
Step #1 - "run-tests": Step #7 - "run-tests":       to equal
Step #1 - "run-tests": Step #7 - "run-tests":           <uint64>: 1
Step #1 - "run-tests": Step #7 - "run-tests":   occurred[0m
Step #1 - "run-tests": Step #7 - "run-tests":   [38;5;9mIn [1m[It][0m[38;5;9m at: [1m/workspace/gloo/pkg/bootstrap/leaderelector/leader_test.go:70[0m [38;5;243m@ 03/20/23 23:16:15.934[0m
Step #1 - "run-tests": Step #7 - "run-tests": 
Step #1 - "run-tests": Step #7 - "run-tests":   [38;5;9mFull Stack Trace[0m
Step #1 - "run-tests": Step #7 - "run-tests":     github.com/solo-io/gloo/pkg/bootstrap/leaderelector_test.glob..func2.4()
Step #1 - "run-tests": Step #7 - "run-tests":     	/workspace/gloo/pkg/bootstrap/leaderelector/leader_test.go:70 +0x1c4
```

**Comment by sam-heilbron** [comment]
I expect this to be resolved on 1.14 by the attached PR. We can re-open if we see this flake again

**Comment by sam-heilbron** [comment]
I expect this to be resolved on 1.14 by the attached PR. We can re-open if we see this flake again
